### ISSUE TITLE
polymorphic relation mysql index problem

### ISSUE THREAD

**Issue by samzhao2008** [strongloop/loopback#2381]
Using mysql as datasource, when I define polymorphic relation, and run database migration to create tables, loopback is trying to add index to discriminator. The result is always failed, because it alter the discriminator field to VARCHAR(512), and of cause, "Specified key was too long; max key length is 767 bytes" error is always thrown. 

**MonthlyInvestment.json**

``` json
{
  "name": "MonthlyInvestment",
  "base": "PersistedModel",
  "plural": "monthlyInvests",
  "idInjection": true,
  "options": {
    "validateUpsert": true
  },
  "properties": {
    "monthlyInvestableId": {
      "type": "number",
      "required": true
    },
    "monthlyInvestableType": {
      "type": "string",
      "dataType": "VARCHAR",
      "dataLength": 32,
      "required": true
    },
    "date": {
      "type": "date",
      "required": true,
      "dateType": "date"
    }
  },
  "validations": [],
  "relations": {
    "educationFund": {
      "type": "belongsTo",
      "model": "EducationFund",
      "polymorphic": {
        "foreignKey": "monthlyInvestableId",
        "discriminator": "monthlyInvestableType"
      }
    }
  }
}
```

education-fund.json

``` json
{
  "name": "EducationFund",
  "base": "PersistedModel",
  "idInjection": true,
  "options": {
    "validateUpsert": true
  },
  "mixins": {
    "TimeStamp": true
  },
  "properties": {
    ....
  },
  "validations": [],
  "relations": {
    "monthlyInvest": {
      "type": "hasMany",
      "model": "MonthlyInvestment",
      "polymorphic": "monthlyInvestable"
    }
  }
}

```

This is annoying, no matter how you define the monthlyInvestableType field(limit it's length), loopback always alters this filed to  VARCHAR(512), and build index, and fail...

The following sql query is what I observed during run database auto update.

``` mysql
ALTER TABLE `MonthlyInvestment` CHANGE COLUMN `monthlyInvestableType` `monthlyInvestableType` VARCHAR(512) NULL,
ADD  INDEX `monthlyInvestableType`  (`monthlyInvestableType`)
```

**Comment by loay** [comment]
Hi @samzhao2008 
Can you post a sample repo so I can reproduce the issue? Thanks.

**Comment by samzhao2008** [comment]
Hi @loay 
I think what I posted is clear. Have you tried the polymorphic relation with mysql data source, or you just can not reproduce the issue?

**Comment by loay** [comment]
Hi @samzhao2008 
If I remember correctly, this issue was fixed in version 3 that was released recently. Can you please double check and let me know if the issue still persists?
Thanks.

**Comment by loay** [comment]
Closing issue for now. If the issue is still persisting, please feel free to leave a comment and have another look. Issue was supposed to be fixed in release of version 3.0. Thanks.
